pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,440 @@
+		pemBytes := r.secret.Data[corev1.SSHAuthPrivateKey]
+		signer, err := ssh.ParsePrivateKey(pemBytes)
+		if err != nil {
+			return fmt.Errorf("failed to parse ssh private key: %w", err)
+		}
 
-// Update updates git repo if remote sha has changed
-func (g *git) Update(branch string) (string, error) {
-	if err := g.clone(branch); err != nil {
-		return "", nil
-	}
+		// PublicKeys implements transport.AuthMethod interface
+		r.auth = &plumbingSSH.PublicKeys{
+			User:   "git",
+			Signer: signer,
+		}
 
-	if err := g.reset("HEAD"); err != nil {
-		return "", err
-	}
+		// Create temporary known_hosts file
+		hostsBytes := r.secret.Data["known_hosts"]
+		if len(hostsBytes) > 0 {
+			f, err := os.CreateTemp("", "known_hosts")
+			if err != nil {
+				return fmt.Errorf("failed to create temporar known_hosts file: %w", err)
+			}
+			// Write received knownhosts from kubernetes secret
+			_, err = f.Write(hostsBytes)
+			if err != nil {
+				return fmt.Errorf("failed to write knonw_hosts file: %w", err)
+			}
+			// Create callback from recently created temporary file
+			hostKeyCB, err := plumbingSSH.NewKnownHostsCallback(f.Name())
+			if err != nil {
+				return err
+			}
 
-	commit, err := g.currentCommit()
-	if err != nil {
-		return commit, err
-	}
+			r.auth = &plumbingSSH.PublicKeys{
+				User:                  "git",
+				Signer:                signer,
+				HostKeyCallbackHelper: plumbingSSH.HostKeyCallbackHelper{HostKeyCallback: hostKeyCB},
+			}
 
-	if changed, err := g.remoteSHAChanged(branch, commit); err != nil || !changed {
-		return commit, err
-	}
+			// Clean known_hosts file after setting up the callback
+			err = f.Close()
+			if err != nil {
+				return fmt.Errorf("failed to close known hosts file: %w", err)
+			}
+			err = os.Remove(f.Name())
+			if err != nil {
+				return fmt.Errorf("failed to remove temporar known_hosts file: %w", err)
+			}
+		}
 
-	if err := g.fetchAndReset(branch); err != nil {
-		return "", err
+		return nil
 	}
 
-	return g.currentCommit()
+	// if all else failed, something nasty happened
+	return errors.New("could not set repository credentials")
 }
 
-// Ensure runs git clone, clean DIRTY contents and fetch the latest commit
-func (g *git) Ensure(branch string) error {
-	if err := g.clone(""); err != nil {
-		return err
-	}
-
-	if err := g.reset(branch); err == nil {
-		return nil
+// setRepoOptions assigns the options configured before in credentials.
+// hard-code other needed configurations like Depth for faster cloning.
+func (r *repository) setRepoOptions() {
+	// Clone Options
+	r.cloneOpts.URL = r.URL
+	r.cloneOpts.Depth = 1
+	r.cloneOpts.InsecureSkipTLS = r.insecureTLSVerify
+	r.cloneOpts.Tags = gogit.NoTags
+
+	// Fetch Options
+	r.fetchOpts.RemoteURL = r.URL
+	r.fetchOpts.Depth = 1
+	r.fetchOpts.Force = true
+	r.fetchOpts.InsecureSkipTLS = r.insecureTLSVerify
+	r.fetchOpts.Tags = gogit.NoTags
+
+	// Checkout Options
+	r.checkoutOpts.Force = true
+	r.checkoutOpts.Create = true
```

Comment:
This is ununsed and so should be removed

Reply:
I have implemented a fix, please resolve the conversation